client/mainwindow.cpp

```diff
@@ -760,7 +761,26 @@ void MainWindow::showLoginWindow(const QString& statusMessage)
         account.sync();
 
         showFirstSyncIndicator();
-        addConnection(connection, dialog.deviceName());
+
+        for (auto a: connections)
```

Comment:
Ideally, instead of an explicit `for()` it would be great to do a `std::find()`:
```suggestion
        const auto it = std::find(connections.cbegin(), connections.cend(),
                [a] (Connection* c) { return a->userId() == c->userId(); });
        if (it != connections.cend())
```
If you hate STL algorithms, at least enclose `connections` into `qAsConst` to prevent COW-detaching (see, e.g., https://stackoverflow.com/questions/35811053/using-c11-range-based-for-loop-correctly-in-qt).